### PR TITLE
restic: Patch restic to remove sha256-simd

### LINKED ISSUE #389: restic: FIPS compatibility w/ restic & minio
**Describe the bug**
The restic binary panics when accessing MinIO (during the mover's initial repo probe). It appears to be trying to use an unsupported (by goboring) crypto algorithm

**Steps to reproduce**

- Run any of the Ansible e2e tests using a FIPS cluster w/ a FIPS enabled restic mover image

**Expected behavior**

**Actual results**
Tests fail

Mover log:
```
        "Starting container",
        "VolSync restic container version: (unknown)",
        "backup",
        "Testing mandatory env variables",
        "== Checking directory for content ===",
        "== Initialize Dir =======",
        "panic: goboringcrypto: hmac hash not recognized",
        "",
        "goroutine 21 [running]:",
        "crypto/hmac.New(0x8?, {0xc0001f0a80?, 0x0?, 0x8?})",
        "    crypto/hmac/hmac.go:136 +0x229",
        "github.com/minio/minio-go/v7/pkg/signer.sumHMAC({0xc0001f0a80?, 0xc0001f0a78?, 0x8?}, {0xc0001f0a90, 0x8, 0x8})",
        "    github.com/minio/minio-go/v7@v7.0.14/pkg/signer/utils.go:40 +0x4a",
        "github.com/minio/minio-go/v7/pkg/signer.getSigningKey({0xc00003a016?, 0x3?}, {0xe9ec5b, 0x9}, {0x9?, 0xc00019a480?, 0x0?}, {0xe98422, 0x2})",
        "    github.com/minio/minio-go/v7@v7.0.14/pkg/signer/request-signature-v4.go:68 +0xfc",
        "github.com/minio/minio-go/v7/pkg/signer.signV4({{0xe986bb, 0x3}, 0xc000284480, {0xe9d11d, 0x8}, 0x1, 0x1, 0xc0001f9e00, {0x0, 0x0}, ...}, ...)",
        "    github.com/minio/minio-go/v7@v7.0.14/pkg/signer/request-signature-v4.go:289 +0x55d",
        "github.com/minio/minio-go/v7/pkg/signer.SignV4(...)",
        "    github.com/minio/minio-go/v7@v7.0.14/pkg/signer/request-signature-v4.go:317",
        "github.com/minio/minio-go/v7.Client.getBucketLocationRequest({0xc000284240, 0xc000122cc0, 0x0, {{0x0, 0x0}, {0x0, 0x0}}, 0x0, 0xc0001f9c80, 0xc000124be0, ...}, ...)",
        "    github.com/minio/minio-go/v7@v7.0.14/bucket-cache.go:251 +0x770",
        "github.com/minio/minio-go/v7.Client.getBucketLocation({0xc000284240, 0xc000122cc0, 0x0, {{0x0, 0x0}, {0x0, 0x0}}, 0x0, 0xc0001f9c80, 0xc000124be0, ...}, ...)",
        "    github.com/minio/minio-go/v7@v7.0.14/bucket-cache.go:100 +0x118",
        "github.com/minio/minio-go/v7.Client.newRequest({0xc000284240, 0xc000122cc0, 0x0, {{0x0, 0x0}, {0x0, 0x0}}, 0x0, 0xc0001f9c80, 0xc000124be0, ...}, ...)",
        "    github.com/minio/minio-go/v7@v7.0.14/api.go:766 +0xde",
        "github.com/minio/minio-go/v7.Client.executeMethod({0xc000284240, 0xc000122cc0, 0x0, {{0x0, 0x0}, {0x0, 0x0}}, 0x0, 0xc0001f9c80, 0xc000124be0, ...}, ...)",
        "    github.com/minio/minio-go/v7@v7.0.14/api.go:637 +0x458",
        "github.com/minio/minio-go/v7.Client.listObjectsV2Query({0xc000284240, 0xc000122cc0, 0x0, {{0x0, 0x0}, {0x0, 0x0}}, 0x0, 0xc0001f9c80, 0xc000124be0, ...}, ...)",
        "    github.com/minio/minio-go/v7@v7.0.14/api-list.go:206 +0x7b4",
        "github.com/minio/minio-go/v7.Client.listObjectsV2.func1(0xc000122d20)",
        "    github.com/minio/minio-go/v7@v7.0.14/api-list.go:98 +0x24e",
        "created by github.com/minio/minio-go/v7.Client.listObjectsV2",
        "    github.com/minio/minio-go/v7@v7.0.14/api-list.go:92 +0x2ba",
        "ERROR: failure checking existence of repository"
```
**Additional context**

- This is known to affect the current main branch, but **we also need to figure out if this affects release-0.5**
- I'm not sure why this has only showed up now... new builder image? or were we not properly testing the restic mover?

### PR DESCRIPTION
**Pull request by JohnStrunk**
**Describe what this PR does**

This adds a script that runs during the container build and removes references to the "optimized" sha256-simd library that was interfering w/ FIPS-compatible compilation

**Is there anything that requires special attention?**

**Related issues:**
Fixes #389

### CODE REVIEW

**Comment by JohnStrunk** [comment]
/cc @tesshuflower 

Are you ok w/ this approach to patching the sources? Since it's just a search/replace at this point, it seemed easiest to just do it in a script w/ the steps outlined.

If it gets more complicated, we can either have patch files (but that only tells you what the changes were, not how to recreate them on a different branch/release) checked in here, or we can fork restic and minio-go into backube and maintain branches w/ the customizations. I don't like either option. :man_shrugging: 

I'm holding this PR until the kuttl PR merges, then I'll make sure this passes OCP CI

**Review by tesshuflower — APPROVED** [review]
/lgtm

I like the script approach John, very nice!  It should also be easy to update the midstream cicd dockerfile similarly - and still know what is going on just by looking at the dockerfile (and patch script).

**Pull request merged**
